compiler/rustc_lint/src/levels.rs

```diff
@@ -1081,6 +1072,18 @@ impl<'s, P: LintLevelsProvider> LintLevelsBuilder<'s, P> {
         let (level, src) = self.lint_level(lint);
         struct_lint_level(self.sess, lint, level, src, span, msg, decorate)
     }
+
+    pub fn emit_lint(
```

Comment:
@davidtwco

Reply:
There's also the option to have `emit_lint` and `emit_spanned_lint` as freestanding implementations on `LintLevelsBuilder`, but it would be better to unify them with `LintContext`.

Reply:
I'm a little confused, isn't there already an implementation of `emit_spanned_lint` and `emit_lint` on `LintContext`?

Reply:
There is, but `LintContext` is not implemented for `LintLevelsBuilder`. Implementing it for `LintLevelsBuilder` would require a type with trait `LintPassObject`, but only the traits `EarlyLintPass` and `LateLintPass` implement it.

Reply:
I'm not sure it's that much of a problem to just have it implemented like you have it now. You could add a new trait which has `emit_lint` and `emit_spanned_lint` and implement that for both `LintLevelsBuilder` and `LintContext` but I don't think we'd be gaining much.